src/test/java/com/scottlogic/filters/FilterANDTest.java

```diff
@@ -0,0 +1,92 @@
+package com.scottlogic.filters;
+
+import com.scottlogic.UserPost;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class FilterANDTest {
+
+    UserPost userPost1 = new UserPost("Joe Bloggs",
+            OffsetDateTime.of(2020, 1, 5, 7, 12, 3, 0, ZoneOffset.UTC),
+            "Hello World!", 2);
+
+    UserPost userPost2 = new UserPost("Joe Bloggs",
+            OffsetDateTime.of(2020, 1, 3, 8, 53, 34, 0, ZoneOffset.UTC),
+            "Another example post.", 1);
+
+    UserPost userPost3 = new UserPost("Joane Smith",
+            OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
+            "An example of a post \nwith lines breaks.", 3);
+
+    List<UserPost> userPosts = Arrays.asList(userPost1, userPost2, userPost3);
+
+    @Test
+    public void filter_authorANDDate_postsWithBothReturned() {
+
+        AuthorFilter filterByAuthor = new AuthorFilter("Jo");
+        DatesFilter filterByDates = new DatesFilter(OffsetDateTime.of(2020, 1, 1, 7, 12, 3, 0, ZoneOffset.UTC), OffsetDateTime.of(2020, 1, 4, 7, 12, 3, 0, ZoneOffset.UTC));
+        FilterAND filteringAnd = new FilterAND(filterByDates, filterByAuthor);
```

Comment:
One of the things that we want to try and achieve with unit tests is making sure that the test is very specific to the class that we are testing.  

The way your test is structured here means that we are dependent on `AuthorFilter` & `DateFilter` functioning correctly for the test to pass. So, we are actually testing `AuthorFilter` , `DateFilter` and `FilterAND`  rather than just the `FilterAND` like how we want to.

To get around this, we want to use mocking as a way to simulate the results of that. Perhaps we can have a chat about that after lunch?